pyaedt/modules/PostProcessor.py

```diff
@@ -1449,218 +1683,213 @@ def get_far_field_data(self, expression="GainTotal", setup_sweep_name='', domain
-            ctxt = domain
-        else:
-            ctxt = ["Domain:=", domain]
 
-        if type(expression) is not list:
-            expression = [expression]
-        if not setup_sweep_name:
-            setup_sweep_name = self._parent.nominal_sweep
+class CircuitPostProcessor(PostProcessorCommon, object):
+    """Manages the main AEDT Nexxim postprocessing functions.
 
 
-        if not report_input_type:
-            report_input_type = report_type[self.post_solution_type]
+    .. note::
+   Some functionalities are available only when AEDT is running in the graphical mode.
 
+    Parameters
+    ----------
+    parent:
+        Inherited parent object. The parent object must provide the members
+        `_modeler`, `_desktop`, `_odesign`, and `_messenger`.
 
-        if families_dict is None:
-            families_dict = {"Freq": ["All"]}
+    """
 
-        solution_data = self.get_solution_data_per_variation(report_input_type, setup_sweep_name, ctxt, families_dict, expression)
+    def __init__(self, parent):
+        PostProcessorCommon.__init__(self, parent)
 
-        if not solution_data:
-            print("No Data Available. Check inputs")
-            return False
-        return solution_data
+    def create_ami_initial_response_plot(self, setupname, ami_name, variation_list_w_value, plot_type="Rectangular Plot", plot_initial_response=True, plot_intermediate_response=False, plot_final_response=False,plotname=None):
+        """
+        Creates an AMI Initial Response Plot
 
-    @aedt_exception_handler
-    def create_rectangular_plot(self, expression="dB(S(1,1))", setup_sweep_name='', families_dict={"Freq": ["All"]},
-                                primary_sweep_variable="Freq", context=None, plotname=None,plottype=None):
-        """Create a 2D rectangular plot in AEDT.
 
         Parameters
         ----------
-        expression : str or list, optional
-            One or more formulas to add to the report. The default is value = ``"dB(S(1,1))"``.
-        setup_sweep_name : str, optional
-            Setup name with the sweep. The default is ``""``.
-        families_dict : dict, optional
-            Dictionary of all families including the primary sweep. The default is ``{"Freq": ["All"]}``.
-        primary_sweep_variable : str, optional
-            Name of the primary sweep. The default is ``"Freq"``.
-        context : str, optional
-            The default is ``None``.
-        plotname : str, optional
-            Name of the plot. The default is ``None``. 
+        setupname: str
+            Name of the setup
+        ami_name: str
+            AMI Probe name to use
+        variation_list_w_value: list
+            list of variations with relative values
+        plot_type: str, Default ``"Rectangular Plot"``
+            String containing the report type. Default is ``"Rectangular Plot"``. it can be ``"Data Table"``, ``"Rectangular Stacked Plot"`` and all other valid AEDT Report types
```

Comment:
```suggestion
            String containing the report type. Default is ``"Rectangular Plot"``. It can be ``"Data Table"``,
            ``"Rectangular Stacked Plot"``or any of the other valid AEDT Report types.
```